rfcs/2021-01-08-5843-encdoing-decoding-format-vector.md

```diff
@@ -0,0 +1,95 @@
+
+-  Our internal data model itself (inclusive of our desire to not incidentally create _another_ observability data format)
+-  How we serialize that data for internal use
+-  The api for accessing that data model (e.g. lookups)
+
+These three concerns don't fully enumerate the problem but I think they're a really excellent distillation of what we need to be considering. So, to reiterate - our ideal solution solves for all three but that might just not be possible. While I initially investigated this problem with really only #2 in mind, discussions with folks on the team led me to feeling like a real firecracker of a solution might be possible for both #1 and #2 out box. With the current state of lookups today, I'm less inclined to try to solve #3 immediately and I'm unconvinced that something might provide us a solution to it directly out of box. (Though, with that said, [flatbuffers](https://google.github.io/flatbuffers/) might be an interesting piece of software to take a look at in the future since it provides users the ability to access serialized data without unpacking it despite it not ultimately being my suggested approach.)
+
+It might not be totally clear from the serialization benchmark examples but while protobufs are inherently slow in some respects, they're _still_ more performant than raw JSON implementations and even MssagePack (which I would be willing to bet remains true for CBOR). Any schema based data format is likely going to have better perf than something schemaless. The schema itself is obviously part of the tooling overhead here but I'd argue that if we cared _more_ about tooling overhead than we did about perfomance and sustainability we'd probably be writing this project in something other than Rust.
+
+The truth is that of the formats that have wide adoption, all of them have their pitfalls and while protobufs bring with it some real tangible maintenance burden its still some of the _best_ tools with better integrations and libraries than most of the other things out there.
+
+As an intentionally buried lede - I wrote this in the hopes that folks would share their opinions on the subject so you might be able to find some gaps in theorem here.
+
+## Prior Art
+
+As mentioned above - my suggestion for the  data spec itself is that we _do_ in fact follow along with prior art. Should we follow this path, we should be giving `Open Telemetry` a thorough exploration. Per the format itself - I haven't found any single conglomerate conlusion about  _any_ encoding format. There are arguments for and against any that we could adopt.
+
+
+## Drawbacks
+
+The drawbacks to fully adopting protobufs on top of an OpenTelemetry spec are many, which makes the path no different than any other.
+    - Protobufs are slower than other schema'd data formats
+    - Protobufs have an unignorable maintenance and tooling overhead
```

Comment:
Maybe it's just me, but I haven't felt this. I also feel like our data model doesn't change very frequently such that updates to the protobuf files are rare. See https://github.com/timberio/vector/commits/master/proto/event.proto